### PR TITLE
[flutter_tools] support any as a special web-hostname

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

When "any" is provided as a --web-hostname, serve the app on anyipv4. Ensure that any is not used as a hostname when spawning chrome.

fyi @DanTup

### CODE REVIEW

**Comment by DanTup** [comment]
Awesome, thanks!

I tested this out on GitPod, but needed to make one additional change so that the DWDS services also bind to the network IPs:

Screenshot 2020-04-14 at 10 31 42

With this PR and that change, I can debug on GitPod without any additional hacks in Flutter (though there are still some VS Code extension / GitPod bugs that need working out).

**Comment by DanTup** [comment]
LGTM, thanks!

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -159,7 +159,11 @@ abstract class FlutterCommand extends Command<void> {
   void usesWebOptions({ bool hide = true }) {
     argParser.addOption('web-hostname',
       defaultsTo: 'localhost',
-      help: 'The hostname to serve web application on.',
+      help:
+        'The hostname that the web sever will use to resolve an IP to serve '
+        'from. The unresolved hostname is used to launch Chrome when using '
+        'the chrome Device. The name "any" may also be used to server on any '
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart:165]: spelling `server -> serve`

**Review by christopherfujino — APPROVED** [review]
LGTM with one comment nit.

**Pull request merged**